File: Python/arranging_coins.py
# https://leetcode.com/problems/arranging-coins/description/
from utils.examination import Examiner, Testcase, Logger


class Solution:
    def arrangeCoins(self, n: int) -> int:
        t: int = 0
        for i in range(1, n + 1):
            t += i
            if t > n:
                return i - 1
            if t == n:
                return i

def examine() -> None:
    Examiner(Solution(), 'arrangeCoins').run((
        Testcase((1,), 1),
        Testcase((2,), 1),
        Testcase((5,), 2),
        Testcase((6,), 3),
        Testcase((8,), 3),
        Testcase((10,), 4),
    ))

File: TypeScript/solutions/arranging-coins.ts
// https://leetcode.com/problems/arranging-coins/description/
import { Examiner, Testcase } from '../utils';


function arrangeCoins(n: number): number {
    let t = 0;
    for (let i = 1; i <= n; i++) {
        t += i;
        if (t > n) return i - 1;
        if (t === n) return i;
    }
    return 0;
}

export function examine(): void {
    new Examiner(arrangeCoins).run([
        new Testcase([1], 1),
        new Testcase([2], 1),
        new Testcase([5], 2),
        new Testcase([6], 3),
        new Testcase([8], 3),
        new Testcase([10], 4),
    ]);
}